src/util/db/dbid.h

```diff
@@ -107,21 +101,30 @@ class DbId {
     }
 
 private:
-  static constexpr value_type kInvalidValue = -1;
+  static constexpr int kInvalidValue = -1;
 
 #if QT_VERSION >= QT_VERSION_CHECK(6, 0, 0)
   static const QMetaType kVariantType;
 #else
     static const QVariant::Type kVariantType;
 #endif
 
-    static bool isValidValue(value_type value) {
+  static bool isValidValue(int value) {
         return 0 <= value;
-    }
-
-    static value_type valueOf(QVariant /*pass-by-value*/ variant);
-
-    value_type m_value;
+  }
+
+  static int valueOf(const QVariant& variant) {
+        bool ok;
+        int value = variant.toInt(&ok);
+        if (ok && isValidValue(value)) {
+            return value;
+        }
+        qCritical() << "Invalid database identifier value:"
+                    << variant;
+        return kInvalidValue;
+  }
+
+  int m_value;
```

Comment:
There was dynamic type code used even though the value is not dynamic and will never be. Even if we change that one day, the value is still a fixed type. So in the unlikely even we will ever change the data type we just need too re-implement the QVariant roundtrip. The benefit is, that this save as us from extra checks and fully inlining now. 

By the way, it also helps reading the code because you immediately see that value() returns an int.

Reply:
@Swiftb0y Do you agree with Daniels answer here, or do you request to change this?

Reply:
I don't agree, but I don't care enough to continue fighting over it.

Reply:
My comment is outted. Now the int type is no longer part of the public interfacebofvthe class. This should fully resolve your concerns.